R/mclosest.R

```diff
@@ -24,37 +24,39 @@
 x <- data.frame(a = 1:5, b = 3:7)
 table <- data.frame(c = c(11, 23, 3, 5, 1), d = c(32:35, 45))
 ppm <- 0.5
-#' 
+#'
 
-mclosest <- function(x, 
-                     table, 
-                     ppm = 0, 
+mclosest <- function(x,
+                     table,
+                     ppm = 0,
                      tolerance = Inf) {
   # sanity checks
-  if (is.null(dim(x)))  
+  if (is.null(dim(x)))
     stop("'x' needs to be an array")
   if (is.null(dim(table)))
     stop("'table' needs to be an array")
   if (ncol(x) != ncol(table))
     stop(" 'x' and 'table' need ot have same number of columns")
-  if (!is.matrix(x)) 
+  if (!is.matrix(x))
     x <- as.matrix(x)
-  if (!is.matrix(table)) 
+  if (!is.matrix(table))
     table <- as.matrix(table)
-  if (length(ppm) != ncol(x))
-   ppm <- rep(ppm[1], ncol(x))
-  
+  nr <- nrow(x)
+  nc <- ncol(x)
+  if (length(ppm) != nc)
+   ppm <- rep(ppm[1], nc)
+  if (length(tolerance) != nc)
+   tolerance <- rep(tolerance[1], nc)
+
   # Initialize a vector to store closest row indices
-  closest_indices <- numeric(nrow(x))
-  
-  for (i in 1:nrow(x)) { 
-    x1 <- x[i,, drop=FALSE]
```

Comment:
we can get rid of the matrix generation. In R, matrices are in fact numeric vectors. so, using `rep( .., nrow(x)` does the trick, no need to create a `matrix` first.